apps/api/src/service/application-service.ts

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (c) 2024 The Ontario Institute for Cancer Research. All rights reserved
+ *
+ * This program and the accompanying materials are made available under the terms of
+ * the GNU Affero General Public License v3.0. You should have received a copy of the
+ * GNU Affero General Public License along with this program.
+ *  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT
+ * SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
+ * TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import { and, asc, desc, eq, sql } from 'drizzle-orm';
+import { ApplicationStates } from 'pcgl-daco/packages/data-model/src/types.ts';
+import { applicationContents } from '../db/schemas/applicationContents.ts';
+import { applications } from '../db/schemas/applications.ts';
+import { type PostgresDb } from '../main.ts';
+
+const applicationService = (db: PostgresDb) => ({
+	createApplication: async ({ user_id }: { user_id: string }) => {
+		const newApplication: typeof applications.$inferInsert = {
+			user_id,
+			state: ApplicationStates.DRAFT,
+		};
+
+		try {
+			const newApplicationRecord = await db.insert(applications).values(newApplication).returning();
+			const { id } = newApplicationRecord[0];
+
+			const newAppContents: typeof applicationContents.$inferInsert = {
+				application_id: id,
+				created_at: new Date(),
+				updated_at: new Date(),
+			};
+			const newAppContentsRecord = await db.insert(applicationContents).values(newAppContents).returning();
+			const { id: contentsId } = newAppContentsRecord[0];
+
+			const application = await db
+				.update(applications)
+				.set({ contents: contentsId })
+				.where(eq(applications.id, id))
+				.returning();
+
+			console.log(`Application created with user_id: ${user_id}`);
+
+			return application[0];
+		} catch (err) {
+			console.error(`Error at createApplication with user_id: ${user_id}`);
+			console.error(err);
+			return null;
+		}
+	},
+	findOneAndUpdate: async ({ id, update }: { id: number; update: any }) => {
```

Comment:
Not listed in the ticket but created out of the need to update records in the tests

Reply:
Update/set types are not well documented but finally figured this out